evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -104,4 +107,12 @@ def validate(self, X, y):
                                                              message_code=DataCheckMessageCode.TARGET_BINARY_INVALID_VALUES,
                                                              details={"target_values": unique_values}).to_dict())
 
+        any_neg = not (y > 0).all() if y.dtype in numeric_dtypes else None
+        if any_neg and (self.objective_name in ['Root Mean Squared Log Error', 'Mean Squared Log Error', 'Mean Absolute Percentage Error']):
```

Comment:
Perhaps instead of having a list here, we should add a `positive_only` static attribute to `ObjectiveBase`, default `False`, set to `True` for these objectives. Then here you could do
```python
if any_neg and handle_objective(self.objective_name).positive_only:
```

And we'd get future compatibility for other similar objectives we add!

What do you think? Would be fine to do this in a separate PR and/or file it.

Reply:
@dsherry Good idea, I'll include it here